build/cloudbuild-automation/README.md

```diff
@@ -0,0 +1,67 @@
+### Building with CloudBuild - Automated Builds
+
+This repo shows how you can leverage Google Cloud Build triggers to automate your custom collector builds based on certain events on GitHub (or other similar version control repository hosting service).
+
+After you have followed instructions in this README, you should have the following automation setup - 
+ - You will have an [Artifact Registry](https://cloud.google.com/artifact-registry) setup in your GCP project where all your custom opentelemetry collector docker images will be published. 
+ - You will have the GitHub repository (containing the code to build your custom collector) linked with a [Cloud Build Trigger](https://cloud.google.com/build/docs/triggers) which will be configured to trigger an automated build for a collector image whenever there is a push to the `main` branch of the repository (*this can be updated to track any branch*).
+
+#### Prerequisites
+- Make sure that you have [docker](https://docs.docker.com/engine/install/) installed on your machine.
+    - Also verify that you are able to run docker commands without the need to gain root permissions using sudo. You can follow the [post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) to achieve this.
+- Set the following environment variables 
+    - `PROJECT_ID` with the GCP project ID in which you will submit cloud builds.
+    - `GOOGLE_APPLICATION_CREDENTIALS` with the path to where your GCP credentials are present, most likely, it will be something like `${HOME}/.config/gcloud/application_default_credentials.json`.
+    - You can set them by executing  
+        ```bash
+        export PROJECT_ID=<your-project-id>
+        # This is where the default credentials live, might be different in case your default location is something else
+        export GOOGLE_APPLICATION_CREDENTIALS=${HOME}/.config/gcloud/application_default_credentials.json
+        ```
+
+#### Repository Structure - Cloudbuild-Automation
+
+1. `tf/main.tf` - folder containing Terraform files that will create necessary resources in your GCP account. 
+2. `tf/setup-build-automation.sh` - convenience shell script that runs the terraform script to generate the resources in GCP account. 
+3. `builder-config.yaml` - configuration defining how to build the custom opentelemetry collector. 
+4. `cloudbuild.yaml` - the YAML file containing steps that need to be run in cloud build. The cloudbuild trigger uses this file to build the docker image for the opentelemetry custom collector.
+
+#### How to run this sample
+
+1. `cd` into the `cloudbuild-automation/tf`. 
```

Comment:
it's easier if you present commands like this in a block-quote (makes copy-pasting one click), ie:
```
cd cloudbuild-automation/tf
```

It's also not clear where the user is starting, so this command might not work. Assuming they are already in the `cloudbuild-automation` folder (reading this), the command should just be `cd tf`

Reply:
Yeah the assumption I took was that user was in the `build` folder. I will replace it though - 

> Assuming you are in the root of the repository, move to the terraform folder containing the script. - 
```
cd build/cloudbuild-automation/tf
```